### PR TITLE
Don't treat certain sync errors as meaning we are offline.

### PR DESCRIPTION
**Pull request by aboodman**
Fixes https://github.com/rocicorp/repc/issues/147

### CODE REVIEW

**Comment by aboodman** [comment]
Testing manually...

File: src/replicache.ts [hunk 1]
```diff
@@ -233,6 +233,8 @@ export default class Replicache implements ReadTransaction {
   }
 
   private async _sync(): Promise<void> {
+    var online = true;
```
Comment (arv) [inline src/replicache.ts:236]: [cbb] no `var` please

File: src/replicache.ts [hunk 2]
```diff
@@ -242,17 +244,24 @@ export default class Replicache implements ReadTransaction {
...
+      // b) sync was aborted because one's already in progress
+      // c) oh noes - something unexpected happened
       //
-      // TODO: we can rethrow here once replicache-internal is improved to not
-      // treat offlineness as an error.
+      // Right now, all of these come out as errors. We distinguish (b) with a
+      // hacky string search. (a) and (c) are not distinguishable currently
+      // because repc doesn't provide sufficient information, so we treat all
+      // errors that aren't (b) as (a).
+      if (e.toString().indexOf('JSLogInfo') == -1) {
```
Comment (arv) [inline src/replicache.ts:259]: [cbb] indexOf -> includes

**Review by arv — COMMENTED** [review]
LGTM

File: src/replicache.ts (hunk 1 above)
Comment (aboodman) [reply src/replicache.ts:236]: Looks like one of us fixed this.

File: src/replicache.ts (hunk 2 above)
Comment (aboodman) [reply src/replicache.ts:259]: In upcoming PR.

File: src/replicache.ts (hunk 1 above)
Comment (arv) [reply src/replicache.ts:236]: I'm going to spend a fe minutes tweaking eslint to check for this and also check sample/**

File: src/replicache.ts [hunk 3]
```diff
@@ -242,17 +244,24 @@ export default class Replicache implements ReadTransaction {
...
+        online = false;
+      }
       console.info(`Error: ${e}`);
```
Comment (phritz) [inline src/replicache.ts:262]: I'm not sure I'd say "Error: " here. "Sync returned"?

**Review by phritz — COMMENTED** [review]
LGTM

File: src/replicache.ts (hunk 3 above)
Comment (aboodman) [reply src/replicache.ts:262]: Good point, done.

**Pull request merged**